### ISSUE TITLE
Error on SenderKeyRecord

### ISSUE THREAD

**Issue by JoseHM8A** [Auties00/WhatsappWeb4j#136]
This error is throwing when i receive a message on groups:

```txt
Jun 16, 2022 2:09:29 AM it.auties.whatsapp.binary.BinarySocket$ErrorHandler handleFailure
WARNING: Received status code 400 at MESSAGE(NoSuchElementException: Cannot find state with id 689172243): handling failure
java.util.NoSuchElementException: Cannot find state with id 689172243
	at it.auties.whatsapp.model.signal.sender.SenderKeyRecord.lambda$findStateById$1(SenderKeyRecord.java:44)
	at java.base/java.util.Optional.orElseThrow(Optional.java:403)
	at it.auties.whatsapp.model.signal.sender.SenderKeyRecord.findStateById(SenderKeyRecord.java:44)
	at it.auties.whatsapp.crypto.GroupCipher.decrypt(GroupCipher.java:45)
	at it.auties.whatsapp.binary.BinarySocket$MessageHandler.decodeCipheredMessage(BinarySocket.java:1240)
	at it.auties.whatsapp.binary.BinarySocket$MessageHandler.decodeMessage(BinarySocket.java:1200)
	at it.auties.whatsapp.binary.BinarySocket$MessageHandler.lambda$decode$35(BinarySocket.java:1192)
	at java.base/java.lang.Iterable.forEach(Iterable.java:75)
	at it.auties.whatsapp.binary.BinarySocket$MessageHandler.decode(BinarySocket.java:1192)
	at it.auties.whatsapp.binary.BinarySocket$StreamHandler.digestMessage(BinarySocket.java:527)
	at it.auties.whatsapp.binary.BinarySocket$StreamHandler.digest(BinarySocket.java:514)
	at it.auties.whatsapp.binary.BinarySocket.handleNode(BinarySocket.java:206)
	at java.base/java.lang.Iterable.forEach(Iterable.java:75)
	at it.auties.whatsapp.binary.BinarySocket.onBinary(BinarySocket.java:201)
	at jdk.internal.reflect.GeneratedMethodAccessor234.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.glassfish.tyrus.core.AnnotatedEndpoint.callMethod(AnnotatedEndpoint.java:553)
	at org.glassfish.tyrus.core.AnnotatedEndpoint$WholeHandler$1.onMessage(AnnotatedEndpoint.java:652)
	at org.glassfish.tyrus.core.TyrusSession.notifyMessageHandlers(TyrusSession.java:560)
	at org.glassfish.tyrus.core.TyrusEndpointWrapper.onMessage(TyrusEndpointWrapper.java:800)
	at org.glassfish.tyrus.core.TyrusWebSocket.onMessage(TyrusWebSocket.java:183)
	at org.glassfish.tyrus.core.frame.BinaryFrame.respond(BinaryFrame.java:70)
	at org.glassfish.tyrus.core.ProtocolHandler.process(ProtocolHandler.java:819)
	at org.glassfish.tyrus.client.TyrusClientEngine$TyrusReadHandler.handle(TyrusClientEngine.java:725)
	at org.glassfish.tyrus.container.grizzly.client.GrizzlyClientFilter$ProcessTask.execute(GrizzlyClientFilter.java:452)
	at org.glassfish.tyrus.container.grizzly.client.TaskProcessor.processTask(TaskProcessor.java:91)
	at org.glassfish.tyrus.container.grizzly.client.TaskProcessor.processTask(TaskProcessor.java:68)
	at org.glassfish.tyrus.container.grizzly.client.GrizzlyClientFilter.handleRead(GrizzlyClientFilter.java:248)
	at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:88)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:246)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:178)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:118)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:96)
	at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:51)
	at org.glassfish.grizzly.nio.transport.TCPNIOTransport.fireIOEvent(TCPNIOTransport.java:510)
	at org.glassfish.grizzly.strategies.AbstractIOStrategy.fireIOEvent(AbstractIOStrategy.java:82)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.run0(WorkerThreadIOStrategy.java:83)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.access$100(WorkerThreadIOStrategy.java:34)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy$WorkerThreadRunnable.run(WorkerThreadIOStrategy.java:101)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.doWork(AbstractThreadPool.java:535)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.run(AbstractThreadPool.java:515)
	at java.base/java.lang.Thread.run(Thread.java:833)
Jun 16, 2022 2:09:29 AM it.auties.whatsapp.binary.BinarySocket$ErrorHandler handleFailure

**Comment by Auties00** [comment]
I'm already working on fixing this issue

**Comment by Auties00** [comment]
Should be fixed in version 3.0-RC5. Please let me know

**Comment by JoseHM8A** [comment]
> Should be fixed in version 3.0-RC5. Please let me know
On RC5 throws the same:

WARNING: Received status code 400 at MESSAGE(NoSuchElementException: Cannot find state with id 960992584): handling failure
java.util.NoSuchElementException: Cannot find state with id 960992584
	at it.auties.whatsapp.model.signal.sender.SenderKeyRecord.lambda$findStateById$1(SenderKeyRecord.java:43)
	at java.base/java.util.Optional.orElseThrow(Optional.java:403)

But on RC6, this has been fixed, but, all the messages are sending: "waiting for messages. this can take a while"
